### ISSUE TITLE
Remove wildcard support for Update

### ISSUE THREAD

**Issue by alerickson** [PowerShell/PSResourceGet#1126]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

> From team discussion, part of the problem is Update-PSResource * may update to newer versions that the user doesn't want. For example, if the user is on 1.0.1, they may only want to update to 1.0.x or optionally to 1.x.y. This assumes semver is being respected. So we need to think through more of the bulk update scenario on both what the user can control and what the defaults should be (I would suggest the safest updates by default so only to newest patch version and not even newer minor version). For the 3.0 release, it may make sense to not allow wildcards so that we can have more time to design for 3.1

see: #747 

### Expected behavior

```console
N/A
```

### Actual behavior

```console
N/A
```

### Environment data

```powershell
N/A
```

**Comment by ThomasNieto** [comment]
The comment references `Update` but this issue is about `Uninstall` which is it?

As for Steve's comment, I disagree with that statement. The user decides what they want to have happen. In v2, `Update-Module` allows wildcards and updates to the latest version. Have there been user complaints with that behavior? Then there is the issue with semver, there is no guarantee that modules adhere to it (module authors can use whatever version scheme they want and put breaking changes in the patch field).

In the original RFC it was proposed to have a `-UpdateTo` parameter with `Major`, `Minor`, and `Patch` enum. This allowed users to specify what level of upgrades they wanted, note I didn't agree with having minor as the default value. Why was that not implemented?

> Update-PSResource will update all resources to most recent minor version by default. A -UpdateTo parameter has values MinorVersion (as default), MajorVersion, PatchVersion. MajorVersion allows updating to latest major version. PatchVersion allows updating to latest patch version (e.g. 6.2.1 to 6.2.4). If the installed resource is a pre-release, it will automatically update to latest prerelease or stable version (if available).

One of my use cases is just running `Update-PSResource`, I don't care if I get newer versions that may or may not contain breaking changes on my dev machine as I would like to know about them before deploying.

The thing really comes down to, are you adding value by disabling wildcards or forcing users to implement work arounds to get what they want done? I don't want or need to use `Get-PSResource | select -Unique | Update-PSResource` while today in v2 I can do the same thing with `Update-Module`.

**Comment by alerickson** [comment]
@ThomasNieto, yup I wrote the wrong cmdlet name in the title-- this issue is regarding Update.

>  In v2, Update-Module allows wildcards and updates to the latest version. Have there been user complaints with that behavior? 

I think you bring up a really good point with this.  I'd actually prefer to leave wildcard functionality in and later implement the `-Updateto` parameter (something we've pushed off til post-GA). That gives users the option to update all packages to something specific, but default behavior can be what was in v2 (just the latest version available).  

Any differing thoughts here @SteveL-MSFT ?
